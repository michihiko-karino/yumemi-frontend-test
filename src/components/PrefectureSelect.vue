<script lang="ts">
import { defineComponent } from 'vue';
import { useStore } from '~/store';

export default defineComponent({
  setup() {
    const { state, getPrefectures, updateTotalPopulationDataSets } = useStore();
    getPrefectures();

    return { state, updateTotalPopulationDataSets };
  },
});
</script>

<template>
  <div class="root">
    <h2>都道府県</h2>
    <ul class="select-fields">
      <li v-for="prefecture in state.prefectures!" :key="prefecture.prefCode">
        <input
          :id="`pref-${prefecture.prefCode}`"
          v-model="state.selectedPrefCodes"
          type="checkbox"
          class="checkbox"
          :value="prefecture.prefCode"
          @change="updateTotalPopulationDataSets(prefecture.prefCode)"
        /><label :for="`pref-${prefecture.prefCode}`" class="label">{{
          prefecture.prefName
        }}</label>
      </li>
    </ul>
  </div>
</template>

<style scoped>
.root {
  padding: 0 20px 40px;
}
.select-fields {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  list-style: none;
  padding-inline-start: 0;
}
.checkbox {
  width: 20px;
  height: 20px;
  margin-right: 4px;
  vertical-align: text-bottom;
}
.label {
  font-size: 18px;
  margin-right: 10px;
}
</style>
