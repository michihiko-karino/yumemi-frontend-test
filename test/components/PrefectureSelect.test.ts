import { mount } from '@vue/test-utils';
import PrefectureSelect from '~/components/PrefectureSelect.vue';

const mockedGetPrefectures = vi.fn();
const mockedUpdateTotalPopulationDataSets = vi.fn();

vi.mock('~/store', () => ({
  useStore: vi.fn(() => ({
    state: { prefectures: [{ prefCode: 1, prefName: '北海道' }] },
    getPrefectures: mockedGetPrefectures,
    updateTotalPopulationDataSets: mockedUpdateTotalPopulationDataSets,
  })),
}));

describe('PrefectureSelect', async () => {
  afterEach(() => {
    vi.restoreAllMocks();
  });

  test('snapshot', () => {
    const wrapper = mount(PrefectureSelect);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('マウント後都道府県APIが呼ばれる', () => {
    mount(PrefectureSelect);
    expect(mockedGetPrefectures).toHaveBeenCalledTimes(1);
  });

  test('都道府県を選択するとstoreのupdateTotalPopulationDataSetsメソッドが実行される', () => {
    const wrapper = mount(PrefectureSelect);
    expect(mockedUpdateTotalPopulationDataSets).toHaveBeenCalledTimes(0);
    wrapper.find<HTMLInputElement>('#pref-1').setValue('true');
    expect(mockedUpdateTotalPopulationDataSets).toHaveBeenCalledTimes(1);
  });
});
