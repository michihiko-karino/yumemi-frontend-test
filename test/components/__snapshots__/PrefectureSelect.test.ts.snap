// Vitest Snapshot v1

exports[`PrefectureSelect > snapshot 1`] = `
"<div class=\\"root\\" data-v-b5485516=\\"\\">
  <h2 data-v-b5485516=\\"\\">都道府県</h2>
  <ul class=\\"select-fields\\" data-v-b5485516=\\"\\">
    <li data-v-b5485516=\\"\\"><input id=\\"pref-1\\" type=\\"checkbox\\" class=\\"checkbox\\" value=\\"1\\" data-v-b5485516=\\"\\"><label for=\\"pref-1\\" class=\\"label\\" data-v-b5485516=\\"\\">北海道</label></li>
  </ul>
</div>"
`;
